pkg/daemon/server/daemon_server.go

```diff
@@ -227,6 +236,77 @@ func (ds *daemonServer) newHTTPServer(ctx context.Context, port int, tlsConfig *
 		log.Info("Not enabling pprof debug endpoints")
 	}
 
-	go ds.exposeMetrics(ctx)
 	return &httpServer
 }
+
+// calculate processing lag and watermark_delay to current time using watermark values.
+func (ds *daemonServer) exposeMetrics(ctx context.Context) {
```

Comment:
are we not emitting watermarks for the rest of the vertices?

Reply:
I see that we only have for sources and sinks, did i get that wrong?

Reply:
We emits the diff of the source and sink, which is the lag we show on the ui, also emits source diff from now.

Reply:
Maybe we could emit watermark for each vertex directly...